File: snippets/fsharp/System/NullReferenceException/Overview/example2.fs
module example2

// <Snippet3>
let populateNames (names: ResizeArray<string>) =
    let arrNames =
        [ "Dakota"; "Samuel"; "Nikita"
          "Koani"; "Saya"; "Yiska"; "Yumaevsky" ]
    for arrName in arrNames do
        names.Add arrName

let getData () : ResizeArray<string> =
    null

let names = getData ()
populateNames names

// The example displays output like the following:
//    Unhandled Exception: System.NullReferenceException: Object reference
//    not set to an instance of an object.
//       at Example.PopulateNames(List`1 names)
//       at <StartupCode$fs>.main()
// </Snippet3>

File: snippets/csharp/System/NullReferenceException/Overview/example2.cs
// <Snippet3>
using System.Collections.Generic;

public class NRE2Example
{
    public static void Main()
    {
        List<string> names = GetData();
        PopulateNames(names);
    }

    private static void PopulateNames(List<string> names)
    {
        string[] arrNames = [ "Dakota", "Samuel", "Nikita",
                            "Koani", "Saya", "Yiska", "Yumaevsky" ];
        foreach (string arrName in arrNames)
            names.Add(arrName);
    }

    private static List<string> GetData()
    {
        return null;
    }
}

// The example displays output like the following:
//    Unhandled Exception: System.NullReferenceException: Object reference
//    not set to an instance of an object.
//       at NRE2Example.PopulateNames(List`1 names)
//       at NRE2Example.Main()
// </Snippet3>
